src/renderer/Dataframe.js

```diff
@@ -21,59 +28,55 @@ export default class Dataframe {
         this.metadata = metadata;
         this.propertyID = {}; // Name => PID
         this.propertyCount = 0;
-        if (this.type === 'polygon') {
-            this._aabb = [];
-            geom.forEach(feature => {
-                const aabb = {
-                    minx: Number.POSITIVE_INFINITY,
-                    miny: Number.POSITIVE_INFINITY,
-                    maxx: Number.NEGATIVE_INFINITY,
-                    maxy: Number.NEGATIVE_INFINITY
-                };
-                feature.forEach(polygon => {
-                    const vertices = polygon.flat;
-                    const numVertices = polygon.holes[0] || polygon.flat.length / 2;
-                    for (let i = 0; i < numVertices; i++) {
-                        aabb.minx = Math.min(aabb.minx, vertices[2 * i + 0]);
-                        aabb.miny = Math.min(aabb.miny, vertices[2 * i + 1]);
-                        aabb.maxx = Math.max(aabb.maxx, vertices[2 * i + 0]);
-                        aabb.maxy = Math.max(aabb.maxy, vertices[2 * i + 1]);
-                    }
-                });
-                this._aabb.push(aabb);
-            });
-        } else if (this.type === 'line') {
-            this._aabb = [];
-            geom.forEach(feature => {
-                const aabb = {
-                    minx: Number.POSITIVE_INFINITY,
-                    miny: Number.POSITIVE_INFINITY,
-                    maxx: Number.NEGATIVE_INFINITY,
-                    maxy: Number.NEGATIVE_INFINITY
-                };
-                feature.forEach(line => {
-                    const vertices = line;
-                    const numVertices = line.length;
-                    for (let i = 0; i < numVertices; i++) {
-                        aabb.minx = Math.min(aabb.minx, vertices[2 * i + 0]);
-                        aabb.miny = Math.min(aabb.miny, vertices[2 * i + 1]);
-                        aabb.maxx = Math.max(aabb.maxx, vertices[2 * i + 0]);
-                        aabb.maxy = Math.max(aabb.maxy, vertices[2 * i + 1]);
-                    }
-                });
-                this._aabb.push(aabb);
-            });
+        this._aabb = this._computeAABB(geom, type);
+    }
+
+    get widthScale () {
+        return this.renderer
+            ? (2 / this.renderer.gl.canvas.clientHeight) / this.scale * this.renderer._zoom
+            : 1;
+    }
+
+    _computeAABB (geometry, type) {
+        switch (type) {
+            case 'point':
+                return [];
```

Comment:
Do we need this `[]`? I expect a hard crash if we try to use an AABB list with points